File: Selenium/Project/Activity6.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.chrome.service import Service as ChromeService
from webdriver_manager.chrome import ChromeDriverManager
driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))

# Navigate to the HRM system login page
driver.get("http://alchemy.hguy.co/orangehrm")

# Find the username and password input fields and login button
username_field = driver.find_element(By.ID, "txtUsername")
password_field = driver.find_element(By.ID, "txtPassword")
login_button = driver.find_element(By.ID, "btnLogin")

# Enter the login credentials
username_field.send_keys("orange")
password_field.send_keys("orangepassword123")

# Click the login button
login_button.click()

# Wait for the navigation menu to become visible
wait = WebDriverWait(driver, 10)
menu = wait.until(EC.presence_of_element_located((By.ID, "menu_directory_viewDirectory")))

# Check if the "Directory" menu item is visible and clickable
if menu.is_displayed() and menu.is_enabled():
    # Click on the "Directory" menu item
    menu.click()

    # Wait for the heading of the page to match "Search Directory"
    search_directory_heading = wait.until(EC.visibility_of_element_located((By.XPATH, "//div[@class='head']/h1")))
    if search_directory_heading.text == "Search Directory":
        print("Directory menu item is visible, clickable, and the heading matches 'Search Directory'.")
    else:
        print("Directory menu item is visible and clickable, but the heading does not match 'Search Directory'.")
else:
    print("Directory menu item is not visible or clickable.")

# Close the browser
driver.quit()

File: Selenium/Project/Activity6.java
package HRM;
 import io.github.bonigarcia.wdm.WebDriverManager;
 import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

public class Activity6 {

        WebDriver driver;

        @BeforeTest
        public void setUp() {
            WebDriverManager.chromedriver().setup();
            //Create a new instance of the Firefox driver
            driver = new ChromeDriver();
            // Open the URL
            driver.get("http://alchemy.hguy.co/orangehrm");
        }

        @Test
        public void verifyDirectoryMenuItem() {
            // Login with credentials
            driver.findElement(By.id("txtUsername")).sendKeys("orange");
            driver.findElement(By.id("txtPassword")).sendKeys("orangepassword123");
            driver.findElement(By.id("btnLogin")).click();

            // Locate the navigation menu
            WebElement directoryMenuItem = driver.findElement(By.cssSelector("a#menu_directory_viewDirectory"));

            // Verify that "Directory" menu item is visible
            Assert.assertTrue(directoryMenuItem.isDisplayed(), "Directory menu item is not visible");

            // Verify that "Directory" menu item is clickable
            Assert.assertTrue(directoryMenuItem.isEnabled(), "Directory menu item is not clickable");

            // Click on the "Directory" menu item
            directoryMenuItem.click();

            // Verify that the heading of the page matches "Search Directory"
            String expectedHeading = "Search Directory";
            String actualHeading = driver.findElement(By.cssSelector("div#content h1")).getText();
            Assert.assertEquals(actualHeading, expectedHeading, "Heading of the page doesn't match");

        }

        @AfterTest
        public void tearDown() {
            driver.quit();
        }

}
